script/tasks/diamondSyncWhitelist.sh

```diff
@@ -513,25 +581,123 @@ function diamondSyncWhitelist {
       if [[ ${#TOKEN_CONTRACTS[@]} -gt 0 ]]; then
         if [[ "$ALLOW_TOKEN_CONTRACTS" == "true" ]]; then
           printf '\033[0;33m%s\033[0m\n' "⚠️  [$NETWORK] Token contracts detected but proceeding (ALLOW_TOKEN_CONTRACTS=true)"
-          printf '\033[0;33m%s\033[0m\n' "Token addresses: ${TOKEN_CONTRACTS[@]}"
+          printf '\033[0;33m%s\033[0m\n' "Token addresses: ${TOKEN_CONTRACTS[*]}"
         else
           printf '\033[0;31m%s\033[0m\n' "❌ [$NETWORK] Token contracts detected in new addresses - aborting whitelist sync"
-          printf '\033[0;31m%s\033[0m\n' "Token addresses: ${TOKEN_CONTRACTS[@]}"
+          printf '\033[0;31m%s\033[0m\n' "Token addresses: ${TOKEN_CONTRACTS[*]}"
           echo ""
           printf '\033[0;33m%s\033[0m\n' "💡 To bypass this check, set ALLOW_TOKEN_CONTRACTS=true and run again:"
           echo ""
           {
             echo "[$NETWORK] Error: Token contracts detected in new addresses"
-            echo "[$NETWORK] Token addresses: ${TOKEN_CONTRACTS[@]}"
+            echo "[$NETWORK] Token addresses: ${TOKEN_CONTRACTS[*]}"
           } >> "$FAILED_LOG_FILE"
           return
         fi
       fi
     fi
 
+    # Remove obsolete contract-selector pairs (process removals first)
+    if [[ ${#REMOVED_PAIRS[@]} -gt 0 ]]; then
+      echoSyncStage "----- [$NETWORK] Stage 4a: Removing obsolete contract-selector pairs -----"
+      printf '\033[0;36m%s\033[0m\n' "📊 [$NETWORK] Found ${#REMOVED_PAIRS[@]} pairs to remove (no longer in whitelist files)"
+      echoSyncStep "🔍 [$NETWORK] Preparing removal transaction with ${#REMOVED_PAIRS[@]} pairs"
+      
+      # Prepare arrays for batch removal operation
+      local REMOVE_CONTRACT_ADDRESSES=()
+      local REMOVE_SELECTORS=()
+
+      echoSyncStep "🔄 [$NETWORK] Processing pairs for removal..."
+      for PAIR in "${REMOVED_PAIRS[@]}"; do
+        # Split pair by '|' to get address and selector
+        CHECKSUMMED_ADDRESS="${PAIR%%|*}"
+        SELECTOR="${PAIR#*|}"
+
+        if [[ -n "$SELECTOR" && "$SELECTOR" != "" ]]; then
+          REMOVE_CONTRACT_ADDRESSES+=("$CHECKSUMMED_ADDRESS")
+          REMOVE_SELECTORS+=("$SELECTOR")
+        fi
+      done
+      echoSyncStep "✔️  [$NETWORK] Processed ${#REMOVE_CONTRACT_ADDRESSES[@]} addresses and ${#REMOVE_SELECTORS[@]} selectors for removal"
+
+      # Convert arrays to cast format
+      local REMOVE_CONTRACTS_ARRAY=""
```

Comment:
can we directly store them in the format that cast needs to avoid several iterations?